streaming/src/main/scala/org/apache/spark/streaming/scheduler/ReceiverTracker.scala

```diff
@@ -371,4 +490,59 @@ class ReceiverTracker(ssc: StreamingContext, skipReceiverLaunch: Boolean = false
       logInfo("Sent stop signal to all " + receiverInfo.size + " receivers")
     }
   }
+
+  def initReceiverTrackingInfos(receivers: Seq[Receiver[_]]): Unit = synchronized {
+    for (receiver <- receivers) {
+      receiverPreferredLocations(receiver.streamId) = receiver.preferredLocation
+      receiverTrackingInfos.put(receiver.streamId, ReceiverTrackingInfo(
+        receiver.streamId,
+        ReceiverState.INACTIVE,
+        None,
+        None))
+    }
+  }
+
+  private def getReceiverTrackingInfoMap(): Map[Int, ReceiverTrackingInfo] =
+    receiverTrackingInfosLock.synchronized {
+      // Copy to an immutable Map so that we don't need to use `synchronized` when using it
+      receiverTrackingInfos.toMap
+    }
+
+  private def updateReceiverScheduledLocations(
+      receiverId: Int, scheduledLocations: Seq[String]): Unit =
```

Comment:
I am not sure whether there is a formatting policy about it but I prefer always using `{` and `}` in multiline methods.